packages/clerk-js/src/utils/authenticateWithPopup.ts

```diff
@@ -0,0 +1,77 @@
+import { buildAccountsBaseUrl } from '@clerk/shared/buildAccountsBaseUrl';
+import type { AuthenticateWithPopupParams, AuthenticateWithRedirectParams } from '@clerk/types';
+
+import type { Clerk } from '../core/clerk';
+
+export async function _authenticateWithPopup(
```

Comment:
Curious why we are exporting a function with a leading `_` ?

Reply:
In our `SignIn` and `SignUp` resources, we have a public `authenticateWithPopup` method available at `signIn.authenticateWithPopup`. These methods share code, so that's pulled out into the shared `_authenticateWithPopup` method. I would _love_ a better name if you have any suggestions, but the underscore is to distinguish it from the actual public method.